packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-dynamo-model-resource-generator.ts

```diff
@@ -149,7 +154,9 @@ export class AmplifyDynamoModelResourceGenerator extends DynamoModelResourceGene
   protected createModelTable(scope: Construct, def: ObjectTypeDefinitionNode, context: TransformerContextProvider): void {
     const modelName = def!.name.value;
     const tableLogicalName = ModelResourceIDs.ModelTableResourceID(modelName);
-    const tableName = context.resourceHelper.generateTableName(modelName);
+    const importedTableMap = context.importedAmplifyDynamoDBTableMap ?? {};
+    const isTableImported = importedTableMap[modelName] !== undefined;
```

Comment:
Piling onto my above comments -- here you check for the existence of the importedTableMap, but _don't_ validate the strategy. If we have conditions that can result in inconsistency b/t strategy & mapping, this is a source of potential future bugs.